### PR TITLE
Initialize dynamic config control_frequency_desired field missed

### PR DESCRIPTION
**Pull request by mikepurvis**
@efernandez This fixes the bug we were looking at today; turns out the config structure isn't initialized to the defaults automatically— hilarious.

### CODE REVIEW

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -372,7 +372,7 @@ namespace diff_drive_controller
     setOdomPubFields(root_nh, controller_nh);
 
     // Set dynamic reconfigure params defaults to the static params provided:
-    DiffDriveControllerConfig config;
+    DiffDriveControllerConfig config(config_default_);
```
Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp:375]: The actual problem is a few lines below.

Could you please ~~revert this change and~~ replace https://github.com/clearpathrobotics/ros_controllers/pull/40/files#diff-380f38b77ed7f79a1de5cf7de13f429eL392 with: ?

``` c++
config.control_frequency_desired = control_frequency_desired_;
```

Thanks!

Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp:375]: Well, we should actually keep this commit too. It's better to have the defaults, than random bits from memory.

Looking in the autogenerated header for the cfg, I can see there's no constructor and copy constructor. So it's using the default ones. That explains the constructor itself doesn't initialize any attribute of the class. :sob:

Comment (mikepurvis) [inline diff_drive_controller/src/diff_drive_controller.cpp:375]: Done.

Comment (mikepurvis) [inline diff_drive_controller/src/diff_drive_controller.cpp:375]: Note also that there's a long-standing ticket about this on dynamic_reconfigure: https://github.com/ros/dynamic_reconfigure/issues/33

Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp:375]: Interesting.

Thanks for the fix and your time debugging it!
